src/app/analyzer-highstock/analyzer-highstock.component.ts

```diff
@@ -274,23 +295,27 @@ export class AnalyzerHighstockComponent implements OnInit, OnChanges {
     }
   }
 
-  checkMaxValues(unit, maxValue, yAxesGroups) {
+  calculateDifference(serie, baseMaxValue) {
```

Comment:
This function could simply return a boolean to make your life easier in the rest of the code.  Ideally, this function would consider the full range of currently plotted points. Consider the following alternative that would require an additional parameter, `baseMinValue`:
```
isOverlapSufficient(serie, baseMin, baseMax) {
  const sufficientOverlap = 0.5;
  const level = serie.chartData ? serie.chartData.level : serie.data;
  const newMin = Math.min(...level.map(l => l[1]));
  const newMax = Math.max(...level.map(l => l[1]));
  const baseRange = baseMax - baseMin;
  const newRange = newMax - newMin;
  const baseMaxNewMin = baseMax - newMin;
  const newMaxBaseMin = newMax - baseMin;
  return Math.min(baseRange, newRange, baseMaxNewMin, newMaxBaseMin) / Math.max(baseRange, newRange) >= sufficientOverlap;
}
```
This should help with series like 148018, which would ideally be on a different axis than 150129.

Reply:
That makes sense, thanks!